packages/plugin-ext-vscode/src/browser/plugin-vscode-commands-contribution.ts

```diff
@@ -315,6 +314,34 @@ export class PluginVscodeCommandsContribution implements CommandContribution {
          * Show Opened File in New Window	workbench.action.files.showOpenedFileInNewWindow
          * Compare Opened File With	workbench.files.action.compareFileWith
          */
+
+        // Register built-in language service commands
+        // see https://code.visualstudio.com/api/references/commands
+        // tslint:disable: no-any
+        const instantiationService = monaco.services.StaticServices.instantiationService.get();
```

Comment:
I would avoid to use monaco object directly.
We have a `theia/monaco` package that should be used provide the communication to monaco.

Reply:
Why and how would adding an indirection help? This is clearly dependening on monaco so obfuscating that will only add unnecessary code and make it harder to understand.

Reply:
There is DI objects in `@theia/monaco` 
Here you're asking for services and you're in the need of importing some monaco definition

By using `@theia/monaco` there is no need to add this index.d.ts and tracking of what is used is clearly identified.

Reply:
The fact is there are too many command registries. If each service is managing its way (and internal way) to get a command registry it's not clean.

Reply:
Those commands are internal monaco commands and only added to our command registry through this contribution. I don't want to add them raw to our command registry and then have this contribution adding them once more using the `vscode.`prefix and needed conversion. Maybe I don't understand what you mean, but adding this to @theia/monaco would do nothing other than exposing it through our API. I don't see how this would help. We do it like this in other places as well and especially here we are vscode-specific so directly using monaco makes perfect sense.

Reply:
What would make more sense to me is to move the additional types I added to the monaco.d.ts in @theia/monaco only locally. But we don't do that so it is easier to verify them in one place when we upgrade monaco.

Reply:
@svenefftinge internal commands of monaco are useful for any theia products as it's based on monaco. So I don't see why theia should 'hide' them and only vscode part should do it.

My problem is that the command registry is quite often bypassed in some portion of code then leading to a hardly understandable codebase. (theia commands and monaco commands) 

The argument 'we do in other place' doesn't look good to me. It's not because we're in a vscode extension that we should bypass or not expose or correctly use services.

Reply:
So you say, we should expose the internal monaco commands to our command registry as is (without a label) within `@theia/monaco` and then add them once more with a `vscode.` prefix and conversion here?

Reply:
well, if I read the discussion from one year, there are still a lot of discussions about the command registries, mapping, wrappers, ranges, etc. so something is not clear.

https://spectrum.chat/theia/dev/plugins-two-problems-related-to-command-parameters~11b89e76-1695-4eb1-b73e-49fb62cfb3f4
https://github.com/eclipse-theia/theia/issues/4050

and latest discussion https://github.com/eclipse-theia/theia/pull/5590

so I would fix it for once.

Reply:
I don't think I can or should fix all the confusion in those discussions. 
But again if I change this PR so that 
1) the internal language-specific commands from monaco (i.e. those starting with `_execute`) are registered in our command registry and 
2) those commands are aliased with `vscode._execute...` and proper conversion in `plugin-ext-vscode` 
3) remove the monaco.d.ts
you are ok with that?

Reply:
@benoitf I have updated the code accordingly.

Reply:
@svenefftinge yes I think code is more maintainable

Reply:
@svenefftinge thx a lot